src/Control/Remote/Util.hs

```diff
@@ -54,7 +54,13 @@ data RemoteMonadException = RemoteEmptyException
 instance Exception RemoteMonadException
 
 -- | Can we dynamically extract the 'result' of a functor, without evaluation.
-
 class KnownResult f where
   knownResult :: f a -> Maybe a
   knownResult _ = Nothing
+
+  unitResult :: f a -> Result a
```

Comment:
It feels a bit strange having both `knownResult` and `unitResult` in the class, as they seem to have overlapping goals. Would it perhaps be nicer to have only one method of type `f a -> Result a`, and change the definition of `Result` to something like this?

```haskell
data Result a where
  UnitResult :: Result ()
  MiscResult :: a -> Result a
  UnknownResult :: Result a
```

Or, if you want to cut down on the number of constructors:

```haskell
data Result a where
  KnownResult :: a -> Maybe (a :~: ()) -> Result a
  UnknownResult :: Result a
```